src/components/progress-bar/progress-bar.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    Component,
+    ChangeDetectionStrategy,
+    HostBinding,
+    Input,
+} from 'angular2/core';
+import {isPresent, CONST} from 'angular2/src/facade/lang';
+import {OneOf} from '../../core/annotations/one-of';
+
+
+// TODO(josephperrott): Benchpress tests.
+// TODO(josephperrott): Add ARIA attributes for progressbar "for".
+
+
+/** Display modes of Progress Bar*/
+@CONST()
+class ProgressMode {
+  @CONST() static DETERMINATE = 'determinate';
+  @CONST() static INDETERMINATE = 'indeterminate';
+  @CONST() static BUFFER = 'buffer';
+  @CONST() static QUERY = 'query';
+}
+
+
+/**
+ * <md-progress-bar> component.
+ */
+@Component({
+  selector: 'md-progress-bar',
+  host: {
+    'role': 'progressbar',
+    'aria-valuemin': '0',
+    'aria-valuemax': '100',
+  },
+  templateUrl: './components/progress-bar/progress-bar.html',
+  styleUrls: ['./components/progress-bar/progress-bar.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MdProgressBar {
+  /**
```

Comment:
The thought here was the separation of what description of the variable from the "requirements" of the variable.

Personally I prefer the separation, but if you would like it on one line I can do that.  But as a wise man once told me, "vertical space is free"